### PR TITLE
Add E2E option to patch SHARED_MESH_CONFIG

### PR DESCRIPTION
**Pull request by nshankar13**
**Please provide a description of this PR:**

Part of https://github.com/istio/istio/issues/45023

### CODE REVIEW

File: pkg/test/framework/components/istio/configmap.go [hunk 1]
```diff
@@ -287,7 +287,19 @@ func (mc *meshConfig) UpdateMeshConfig(t resource.Context, update func(*meshconf
 	for _, c := range mc.ctx.AllClusters().Kube() {
 		c := c
```
Comment (keithmattix) [inline pkg/test/framework/components/istio/configmap.go:288]: What about the logic on L242?

Comment (nshankar13) [reply pkg/test/framework/components/istio/configmap.go:293]: Good catch, updated.

File: pkg/test/framework/components/istio/flags.go [hunk 2]
```diff
@@ -65,4 +65,12 @@ func init() {
 		settingsFromCommandline.EgressGatewayIstioLabel,
 		`Specifies the istio label of the egressgateway to search for when running tests in a preinstalled istio installation.
 		Should only be set when istio.test.kube.deploy=false`)
+	flag.StringVar(&settingsFromCommandline.UseSharedMeshConfig, "istio.test.kube.useSharedMeshConfig",
```
Comment (howardjohn) [inline pkg/test/framework/components/istio/flags.go:68]: nit: no need for the flag? `sharedMeshConfigName` being non-empty implies to use it?

**Review by howardjohn — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]

File: pkg/test/framework/components/istio/flags.go [hunk 3]
```diff
@@ -65,4 +65,8 @@ func init() {
 		settingsFromCommandline.EgressGatewayIstioLabel,
 		`Specifies the istio label of the egressgateway to search for when running tests in a preinstalled istio installation.
 		Should only be set when istio.test.kube.deploy=false`)
+	flag.StringVar(&settingsFromCommandline.SharedMeshConfigName, "istio.test.kube.sharedMeshConfigName",
+		settingsFromCommandline.SharedMeshConfigName,
+		`Specifies the name of the SHARED_MESH_CONFIG defined and created by the user upon installing Istio.
+		Should only be set when istio.test.kube.userSharedMeshConfig=true and istio.test.kube.deploy=false.`)
```
Comment (whitneygriffith) [inline pkg/test/framework/components/istio/flags.go:71]: `istio.test.kube.userSharedMeshConfig=true` still needs to be implemented? Or should this be removed?

Comment (Winbobob) [reply pkg/test/framework/components/istio/flags.go:71]: Just checked with Niranjan, `istio.test.kube.userSharedMeshConfig=true` should be removed, just `SharedMeshConfigName` is good enough.

**Pull request merged**